### PR TITLE
Add reporter on postprocessor

### PR DESCRIPTION
**Pull request by colinepiloquet**
Signed-off-by: Coline Piloquet <coline.piloquet@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Adds reporter in the import post-processor

### CODE REVIEW

**Review by miovd — COMMENTED** [review]
Could you add a test? (With a mock ImportPostProcessor)

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/ImportersTest.java [hunk 1]
```diff
@@ -209,4 +215,27 @@ public void loadNetworks() throws InterruptedException, ExecutionException, IOEx
...
+
+        ReporterModel reporter = new ReporterModel("testPostProcessor", "Test importer post processor");
+        Network network1 = importer1.importData(null, NetworkFactory.findDefault(), null, reporter);
+        assertNotNull(network1);
+
+        Optional<Report> report = reporter.getReports().stream().findFirst();
+        assertTrue(report.isPresent());
+
+        StringWriter sw = new StringWriter();
+        reporter.export(sw);
```
Comment (flo-dup) [inline iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/ImportersTest.java:233]: It's more consistent to check the serialization, you can use `ReporterModelSerializer::write` for that.

**Review by flo-dup — CHANGES_REQUESTED** [review]

**Review by flo-dup — APPROVED** [review]

**Comment by miovd** [comment]
@colinepiloquet Could you add a test on `ImportPostProcessor.process(Network network, ComputationManager computationManager)`? It is the one that blocks SonarCloud

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/PostProcessorReporterTest.java [hunk 2]
```diff
@@ -14,27 +14,28 @@
...
 /**
  * @author Coline Piloquet <coline.piloquet at rte-france.com>
  */
 public class PostProcessorReporterTest extends AbstractConverterTest {
+    Importer testImporter = new TestImporter();
+    ImportPostProcessorMock importPostProcessorMock = new ImportPostProcessorMock();
+    ImportersLoader loader = new ImportersLoaderList(Collections.singletonList(testImporter), Collections.singletonList(importPostProcessorMock));
+    ComputationManager computationManager = Mockito.mock(ComputationManager.class);
+    Importer importer1 = Importers.addPostProcessors(loader, testImporter, computationManager, "testReporter");
+
```
Comment (miovd) [inline iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/PostProcessorReporterTest.java:36]: Can those be private and final?

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/PostProcessorReporterTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        Optional<Report> report = reporter.getReports().stream().findFirst();
+        assertTrue(report.isPresent());
+
+        roundTripTest(reporter, ReporterModelSerializer::write, ReporterModelDeserializer::read, "/postProcessorReporterTest.json");
+    }
+
+    @Test
+    public void postProcessorWithoutReporter() {
+        Network network1 = importer1.importData(null, NetworkFactory.findDefault(), null);
```
Comment (miovd) [inline iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/PostProcessorReporterTest.java:52]: I think you have to call explicitly `process(Network, ComputationManager)`

**Pull request merged**